provider/go.mod

```diff
@@ -0,0 +1,126 @@
+module github.com/pulumi/pulumi-eks/provider/v3
+
+go 1.23
+
+toolchain go1.23.4
+
+replace github.com/pulumi/pulumi-eks/sdk/v3 => ../sdk
```

Comment:
I'm curious why/how these Go files depend on the SDK for the EKS provider? Does it not create some circular dependency in the sense? Surprised. I thought all these files do is wrapping schema generation and schema-embed compression.

Reply:
It actually doesn't. The replace directive isn't needed since we aren't actually using `github.com/pulumi/pulumi-eks/sdk/v3 `. It was included as I lifted the go.mod file as is. I can clean this up in a follow-up PR.